### PR TITLE
Create Ship and Cell classes

### PR DESCRIPTION
**Pull request by hillstew**
Ship class w/ following methods:
1. initialize
2. sunk?
3. hit

Cell class w/ following methods:
1. initialize
2. empty
3. place_ship
4. fire_upon
5. fired_upon?
6. render - this was most challenging and would love to hear any feedback you have

Fixed the indentation to use two spaces per tab in the test file.

### CODE REVIEW

File: lib/ship.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  attr_reader :name, :length, :health
+
+  def initialize(name, length)
+    @name = name
+    @length = length
+    @health = length
+  end
+
+  def sunk?
+    @health <= 1
```
Comment (m3gacrypt0) [inline lib/ship.rb:11]: @health <= 0 #health of 1 is not sunk in specifications

File: lib/cell.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  def fire_upon
+    @hit_status = true
+    @ship.hit
+  end
+
+  def fired_upon?
+    @hit_status
+  end
+
+  def render(reveal_ship=false)
```
Comment (m3gacrypt0) [inline lib/cell.rb:27]: `#Not sure if you are hitting all possibilities with the implementation, I would try this:
def render(reveal_ship=true)
  if @cell.fired_upon?
    if @cell.ship != nil && @cell.ship.health > 0
      "H"
    elsif @cell.ship != nil
      "X"
    else
      "M"
    end
  elsif reveal_ship && @cell.ship != nil
    "S"
  else
    "."
  end
end`

**Review by m3gacrypt0 — APPROVED** [review]

File: lib/ship.rb (hunk 1 above)
Comment (hillstew) [reply lib/ship.rb:11]: Ah good catch!

File: lib/cell.rb (hunk 2 above)
Comment (hillstew) [reply lib/cell.rb:27]: Can you slack me this code snippet? GitHub's formatting is awful 😁

**Pull request merged**